TGM/src/main/java/network/warzone/tgm/modules/SpawnPointHandlerModule.java

```diff
@@ -82,13 +86,11 @@ public void spawnPlayer(PlayerContext playerContext, MatchTeam matchTeam, boolea
                 if (gameClassModule.getCurrentClass(playerContext.getPlayer()) == null) gameClassModule.setCurrentClass(playerContext.getPlayer(), gameClassModule.getDefaultClass());
                 if (matchTeam.isSpectator()) {
                     spectatorModule.applySpectatorKit(playerContext);
-                } else {
-                    if (reset) {
-                        gameClassModule.performSwitch(playerContext.getPlayer());
-                        GameClass gameClass = gameClassModule.getGameClass(gameClassModule.getCurrentClass(playerContext.getPlayer()));
-                        if (gameClass != null) gameClass.apply(playerContext.getPlayer(), matchTeam.getColor());
-                        playerContext.getPlayer().updateInventory();
-                    }
+                } else if (reset) {
```

Comment:
FYI you shouldn't touch unrelated code but this looks okay.

Reply:
Alright, I'll make sure to make separate PRs for cleaning up unrelated code in the future